src/generators/Silk.NET.SilkTouch.Common/Generation/SilkTouchContext.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Silk.NET.SilkTouch.Configuration;
+
+namespace Silk.NET.SilkTouch.Generation
+{
+    public sealed record SilkTouchContext
+    (
+        string AssemblyName,
+        IEnumerable<CSharpSyntaxTree> SyntaxTrees,
```

Comment:
I need a compilation. Without a Compilation Overloads & Emitter probably won't work

Reply:
What specifically do you need from the compilation though?

Reply:
Ok fair enough.